### PR TITLE
TASK: Refresh node list after a node is created

### PR DESCRIPTION
**Pull request by Sebobo**
Resolves: #5

### CODE REVIEW

File: Resources/Private/FlatNav/src/FlatNav.js [hunk 1]
```diff
@@ -181,6 +182,19 @@ class FlatNav extends Component {
...
+
+    handleNodeWasCreated = (feedbackPayload, {store}) => {
+        const state = store.getState();
+
+        const getNodeByContextPathSelector = selectors.CR.Nodes.makeGetNodeByContextPathSelector(feedbackPayload.contextPath);
+        const node = getNodeByContextPathSelector(state);
+        const nodeTypeName = $get('nodeType', node);
+
+        if (nodeTypeName === this.props.preset.newNodeType) {
+            this.refreshFlatNav();
```
Comment (dimaip) [inline Resources/Private/FlatNav/src/FlatNav.js]: Too bad that it does a full refresh.
Wouldn't it be better to just append the new node to the top of the list?

Comment (dfeyer) [reply Resources/Private/FlatNav/src/FlatNav.js]: Let the optimization for later, so tree can be sorted or even grouped in the future and this kind of early optimization can block us later.

Comment (dimaip) [reply Resources/Private/FlatNav/src/FlatNav.js]: I'm not sure this extension should solve all of the possible usecases, it was originally meant to show how easy it is to create custom UI solutions targeted at a specific usecase...
Besides, if we would choose to do something more advanced later we could always remove that optimization.
But anyways, I'm OK to leave it as is. Let @Sebobo decide and merge as he pleases.

Comment (dimaip) [reply Resources/Private/FlatNav/src/FlatNav.js]: Agreed. Just FYI, to put manually you'd just need to get the new node information from the feedback and add it to state, in the same way you did when loading nodes from load more button.

**Pull request merged**